bootstrap/main.go

```diff
@@ -0,0 +1,209 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+
+	flag "github.com/spf13/pflag"
+)
+
+func printUsage(flags *flag.FlagSet) {
+	log.Println("Flags")
+	log.Println(flags.FlagUsages())
+}
+
+// ParseArgs parses the command line to supported arguments as a pflag.FlagSet
+// arguments should be generally be os.Args[1:]
+func ParseArgs(arguments []string) *flag.FlagSet {
+	args := flag.NewFlagSet("bootstrap", flag.ExitOnError)
```

Comment:
`flag.Bool()` or `flag.BoolVar()` seem more common patterns in this org. Any reason you're not using them here?
https://github.com/kubernetes/test-infra/blob/master/kubetest/main.go#L109

Reply:
If I change the exit behavior here having a `NewFlagSet` will make unit testing easier, using bare `flag.X` just uses the default `FlagSet` inside `pflag`, and I don't see any way to reset / clear the `FlagSet` for parsing again in tests.

Reply:
I moved the parsed args to a struct and switched to the `*Var` variants instead of passing around the `FlagSet` to be a bit more consistent with the rest of k8s.